daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SValue.scala

```diff
@@ -234,7 +230,11 @@ object SValue {
     def bool(b: Boolean) = if (b) True else False
   }
 
-  private val entryFields = ImmArray(Ast.keyFieldName, Ast.valueFieldName)
+  private val entryFields = Struct.assertFromSeq(List(keyFieldName -> (()), valueFieldName -> (())))
+
+  // we verify the fields are ordered as the `entry` method expects it.
+  assert(entryFields.indexOf(keyFieldName) == 0)
```

Comment:
Why the assertion here instead of changing `entry` to not care about the ordering like you did in other places?

Reply:
I wanted to avoid dynamic check because this is used by the built-in `TEXTMAP_TO_LIST`.

Reply:
I wasn’t thinking of a check. Currently you have
```
  private def entry(key: SValue, value: SValue) = {
    val args = new util.ArrayList[SValue](2)
    args.add(key)
    args.add(value)
    SStruct(entryFields, args)
  }
```
If you change that to
```
  private def entry(key: SValue, value: SValue) = {
    val args = new util.ArrayList[SValue](2)
    args.set(indexOfKeyField, key)
    args.set(indexOfValueField, value)
    SStruct(entryFields, args)
  }
```
However, I just realized that this will of course throw an out of range and I couldn’t find a method to initialize an ArrayList with `size` 2 instead of `capacity` 2 so ignore my comment.